x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -255,4 +256,15 @@ static DeprecationIssue checkRemovedSetting(final Settings settings, final Setti
         return new DeprecationIssue(DeprecationIssue.Level.CRITICAL, message, url, details);
     }
 
+    static DeprecationIssue javaVersionCheck(Settings nodeSettings, PluginsAndModules plugins) {
+        if (JavaVersion.current().compareTo(JavaVersion.parse("11")) < 0) {
+            return new DeprecationIssue(DeprecationIssue.Level.CRITICAL,
+                "Java 11 is required",
+                "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-8.0.html#_java_11_is_required",
+                "Java 11 will be required for future versions of Elasticsearch, this node is running version "
```

Comment:
I wonder if we should first suggest using the bundled jdk?

Reply:
Good idea. I've added a note here, and in a couple other places too. What do you think?

Reply:
The new wording is better, but it may be confusing for some users since they may have the bundled jdk, but have JAVA_HOME set by their system, so what they need to do is unset JAVA_HOME.

Reply:
I've expanded the message further.